src/app/public/modules/adapter-service/adapter.service.spec.ts

```diff
@@ -186,4 +191,80 @@ describe('Core adapter service', () => {
       expect(container).toHaveCssClass('sky-responsive-container-lg');
     });
   });
+
+  describe('isTargetAboveElement', () => {
+    let adapter: SkyCoreAdapterService;
+    let container: HTMLDivElement;
+
+    beforeEach(inject(
+      [SkyCoreAdapterService], (_adapter: SkyCoreAdapterService) => {
+        adapter = _adapter;
+        container = document.getElementById('z-index-container') as HTMLDivElement;
+      }
+    ));
+
+    it('should check if event target is above element', () => {
+        const div1 = document.createElement('div');
+        div1.style.position = 'fixed';
+        div1.style.zIndex = '1';
+
+        const div2 = document.createElement('div');
+        div2.style.position = 'fixed';
+        div2.style.zIndex = '2';
+
+        container.appendChild(div1);
+        container.appendChild(div2); // Higher z-index added last.
+
+        const result = adapter.isTargetAboveElement(div2, div1);
+        expect(result).toEqual(true);
+    });
+
+    it('should handle elements that do not exist in the DOM', () => {
+      const div1 = document.createElement('div');
+      div1.style.position = 'fixed';
+      div1.style.zIndex = '1';
+
+      const div2 = document.createElement('div');
+      div2.style.position = 'fixed';
+      div2.style.zIndex = '2';
+
+      // Only add the first element.
+      container.appendChild(div1);
+
+      const result = adapter.isTargetAboveElement(div2, div1);
+      expect(result).toEqual(true);
+    });
+
+    it('should walk parents until z-index is found', () => {
```

Comment:
"should walk"? Not sure what that means here...

Reply:
"Walking an array" is another way of saying "iterating over an array of ancestors recursively". Python and PHP use the term often. If you hate it, we can change it.